integration/benchmark/main.go

```diff
@@ -0,0 +1,353 @@
+	"strconv"
+	"strings"
+	"time"
+
+	flowsdk "github.com/onflow/flow-go-sdk"
+	client "github.com/onflow/flow-go-sdk/access/grpc"
+
+	"github.com/onflow/flow-go/cmd/build"
+	"github.com/onflow/flow-go/crypto"
+	"github.com/onflow/flow-go/integration/utils"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/metrics"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+type LoadCase struct {
+	tps      int
+	duration time.Duration
+}
+
+// This struct is used for uploading data to BigQuery, changes here should
+// remain in sync with tps_results_schema.json
+type dataSlice struct {
+	GoVersion           string
+	OsVersion           string
+	GitSha              string
+	StartTime           time.Time
+	EndTime             time.Time
+	InputTps            float64
+	OutputTps           float64
+	ProStartTransaction float64
+	ProEndTransaction   float64
+}
+
+func main() {
+	// holdover flags from loader/main.go
+	logLvl := flag.String("log-level", "info", "set log level")
+	profilerEnabled := flag.Bool("profiler-enabled", false, "whether to enable the auto-profiler")
+	maxConstExecTxSizeInBytes := flag.Uint("const-exec-max-tx-size", flow.DefaultMaxTransactionByteSize/10, "max byte size of constant exec transaction size to generate")
+	authAccNumInConstExecTx := flag.Uint("const-exec-num-authorizer", 1, "num of authorizer for each constant exec transaction to generate")
+	argSizeInByteInConstExecTx := flag.Uint("const-exec-arg-size", 100, "byte size of tx argument for each constant exec transaction to generate")
+	payerKeyCountInConstExecTx := flag.Uint("const-exec-payer-key-count", 2, "num of payer keys for each constant exec transaction to generate")
+
+	// CI relevant flags
+	tpsFlag := flag.String("tps", "300", "transactions per second (TPS) to send, accepts a comma separated list of values if used in conjunction with `tps-durations`")
+	tpsDurationsFlag := flag.String("tps-durations", "10m", "duration that each load test will run, accepts a comma separted list that will be applied to multiple values of the `tps` flag (defaults to infinite if not provided, meaning only the first tps case will be tested; additional values will be ignored)")
+	ciFlag := flag.Bool("ci-run", false, "whether or not the run is part of CI")
+	leadTime := flag.String("leadTime", "30s", "the amount of time before data slices are started")
+	sliceSize := flag.String("sliceSize", "2m", "the amount of time that each slice covers")
+	flag.Parse()
+
+	// Hardcoded CI values
+	loadType := "token-transfer"
+	access := net.JoinHostPort("127.0.0.1", "3569")
+	metricport := uint(8080)
+	pushgateway := "127.0.0.1:9091"
+	accountMultiplier := 50
+	feedbackEnabled := true
+	serviceAccountPrivateKeyHex := unittest.ServiceAccountPrivateKeyHex
+
+	// Version and Commit Info
+	gitSha := build.Commit()
+	goVersion := runtime.Version()
+	osVersion := runtime.GOOS + runtime.GOARCH
+
+	chainID := flowsdk.Emulator
+
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
my comment in the other file still applies.  this context does nothing since it's only cancel at the very end.  just remove it and just run the goroutine (line 173) body as part of main

Reply:
Completed in both files.